style: Add animation to `Tabs` indicator

fartinmartin opened the pull request:
## What changed?
Fixes #304

## How was this change made?
This PR uses `TabListStateContext` within `TabList` to set `--count` and `--active` CSS variables. These variables are used in `Tab.module.css` to calculate the size and position of the `.tabs-list::after` pseudo element using the CSS `calc()` function.

Other methods were considered (using `:has()` and/or CSS view transitions), but I think this PR is a more robust solution. For prosperity sake, https://github.com/fartinmartin/namesake/commit/45003c570a09f39bd7a8c0368298a7fd23a2c227 has both of my explorations. Head over to `/signin` and comment out the relevant import statements to compare them.

## How was this tested?
Mostly by spamming `/signin`, but also clicking around after running `pnpm storybook`. I also tested with `<Tabs ... orientation="vertical">`

## Anything else?
While I was attempting to commit I was getting a type error in `src/utils/encryption.ts` which I hadn't edited—strange! So, this PR currently has this unrelated edit: https://github.com/fartinmartin/namesake/commit/fd6d5ea41d261517d73d4a2c28b87383c1d59a9e

evadecker commented on src/components/common/Tabs/Tabs.tsx:
```diff
@@ -44,23 +47,37 @@ const tabListStyles = tv({
 const tabProps = tv({
   extend: focusRing,
-  base: "flex items-center justify-center cursor-pointer rounded-md px-4 py-2 text-sm transition forced-color-adjust-none",
+  base: "flex items-center justify-center cursor-pointer px-4 py-2 text-sm transition forced-color-adjust-none",
```
We'll want to keep `rounded-md` here for the focus ring to remain rounded:

![CleanShot 2025-03-11 at 10 34 23@2x](https://github.com/user-attachments/assets/e37a9300-7092-4d95-89ea-0231d818d92c)

fartinmartin replied:
Oh, good catch!

evadecker commented on src/components/common/Tabs/Tabs.stories.tsx:
```diff
@@ -1,5 +1,5 @@
-import { Tab, TabList, TabPanel, Tabs } from "@/components/common";
 import type { Meta } from "@storybook/react";
+import { Tab, TabList, TabPanel, Tabs } from "./Tabs";
```
Noticing that in dark mode in Storybook, the text doesn't render properly. Wonder if it's worth setting the base text color for the `Tabs` component?

![CleanShot 2025-03-11 at 10 35 04@2x](https://github.com/user-attachments/assets/af7cfd91-1e23-4b29-8b88-d3eceda86868)

evadecker replied:
On reflection this appears to be a problem with all Storybook stories! Disregard :)

evadecker commented on src/components/common/Tabs/Tabs.module.css:
```diff
@@ -0,0 +1,41 @@
+
+.tab-list::after {
+  /* Tailwind's `after:transition` class is not working for me :/ */
+  transition: translate 0.2s ease;
```
```suggestion
  transition: translate 0.2s ease-out;
```
Personal preference: `ease-out` is nice for transitions since it's fast at the start (reacts quickly) and slow at the end (gradually finds its place)

fartinmartin replied:
Smart!

evadecker reviewed (commented):
Hey! Sorry it took so long to add a review here. This is looking really good! Animates smoothly. :)

I left a few comments, let me know if you have any questions.

fartinmartin commented on src/components/common/Tabs/Tabs.module.css:
```diff
@@ -0,0 +1,41 @@
+.tab-list {
+  /* [tabListStyles.base] n/a */
+  --tablist-gap-x: 0;
+  --tablist-gap-y: 0;
+  /* [tabListStyles.base] p-1 */
+  --tablist-padding-x: 0.25rem;
+  --tablist-padding-y: 0.25rem;
+  /* [tabProps.base] px-4 py-2 */
+  --tab-padding-x: 1rem;
+  --tab-padding-y: 0.5rem;
+
+  --count: 1; /* updated in Tabs.tsx > TabList */
+  --active: 0; /* updated in Tabs.tsx > TabList  */
```
Thank you!

fartinmartin commented on tests/e2e/signin.spec.ts:
```diff
@@ -0,0 +1,54 @@
+// import { faker } from "@faker-js/faker";
```
Oop, I did not!

fartinmartin commented on tsconfig.app.json:
```diff
@@ -1,8 +1,8 @@
 {
   "compilerOptions": {
-    "target": "ES2020",
+    "target": "ESNext",
```
Strange, not sure how/why these were changed!

fartinmartin commented:
Hey @evadecker, no worries at all—thanks for taking a look! Sorry _I_ was slow to respond! I've addressed your comments, let me know if you have any other notes!
